src/main/java/org/folio/rest/core/RestClient.java

```diff
@@ -1,191 +1,265 @@
+
+    var recordData = JsonObject.mapFrom(dataObject);
+    if (log.isDebugEnabled()) {
+      log.debug(REQUEST_MESSAGE_LOG_DEBUG, HttpMethod.PUT, JsonObject.mapFrom(recordData).encodePrettily());
     }
+    var caseInsensitiveHeader = convertToCaseInsensitiveMap(requestContext.getHeaders());
 
-    return future;
+    return getVertxWebClient(requestContext.getContext())
+      .putAbs(buildAbsEndpoint(caseInsensitiveHeader, endpoint))
+      .putHeaders(caseInsensitiveHeader)
+      .expect(SUCCESS_RESPONSE_PREDICATE)
+      .sendJson(recordData)
+      .onFailure(log::error)
+      .mapEmpty();
   }
 
-  public <T> CompletableFuture<Void> put(RequestEntry requestEntry, T entity, RequestContext requestContext) {
-    CompletableFuture<Void> future = new CompletableFuture<>();
+  public <T>Future<Void> patch(RequestEntry requestEntry, T dataObject, RequestContext requestContext ) {
     String endpoint = requestEntry.buildEndpoint();
-    JsonObject recordData = JsonObject.mapFrom(entity);
+    return patch(endpoint, dataObject, requestContext);
+  }
 
-    if (logger.isDebugEnabled()) {
-      logger.debug("Sending 'PUT {}' with body: {}", endpoint, recordData.encodePrettily());
+  public <T>Future<Void> patch(String endpoint, T dataObject, RequestContext requestContext) {
+    log.info(REQUEST_MESSAGE_LOG_INFO, HttpMethod.PATCH, endpoint);
+    var recordData = JsonObject.mapFrom(dataObject);
+    if (log.isDebugEnabled()) {
+      log.debug(REQUEST_MESSAGE_LOG_DEBUG, HttpMethod.PUT, JsonObject.mapFrom(recordData).encodePrettily());
     }
+    var caseInsensitiveHeader = convertToCaseInsensitiveMap(requestContext.getHeaders());
 
-    HttpClientInterface client = getHttpClient(requestContext.getHeaders());
-    setDefaultHeaders(client);
-    try {
-      client
-          .request(HttpMethod.PUT, recordData.toBuffer(), endpoint, requestContext.getHeaders())
-          .thenAccept(HelperUtils::verifyResponse)
-          .thenAccept(avoid -> {
-            client.closeClient();
-            future.complete(null);
-          })
-          .exceptionally(t -> {
-            client.closeClient();
-            future.completeExceptionally(t.getCause());
-            logger.error("'PUT {}' request failed. Request body: {}", endpoint, recordData.encodePrettily(), t.getCause());
-            return null;
-          });
-    } catch (Exception e) {
-      logger.error("'PUT {}' request failed. Request body: {}", endpoint, recordData.encodePrettily(), e);
-      client.closeClient();
-      future.completeExceptionally(e);
-    }
+    Promise<Void> promise = Promise.promise();
 
-    return future;
+    return getVertxWebClient(requestContext.getContext())
+      .patchAbs(buildAbsEndpoint(caseInsensitiveHeader, endpoint))
+      .putHeaders(caseInsensitiveHeader)
+      .expect(SUCCESS_RESPONSE_PREDICATE)
+      .sendJson(dataObject)
+      .onSuccess(json -> promise.complete())
+      .onFailure(log::error)
```

Comment:
Seems promise will be never completed in case of failure

Reply:
Actually it does not affect anything. However, I removed patch method at all as not used anywhere